### DISCUSSION TITLE
A problem with finding Vivado Path

### DISCUSSION THREAD

**Discussion by MehdiRh17** [Xilinx/finn#529] [Q&A] [answered]
Hello. I am trying to setup Xilinx'Finn examples notebook with docker. after running the first two notebooks in end2endexamples/cybersecurity, I get this error in the third part:

AssertionError: vivado_hls not found in PATH

I have already assigned FINN_XILINX_PATH and FINN_XILINX_VERSION according to the finn.readthedocs.io
I'm on Ubuntu 20.04 and my Vivado is 2019.1. I had this problem before with 2020.2 and I noticed that I should install the 2020.1 or below but I still have the same problem! I really appreciate it if you can help me with this. Thanks

**Comment by nhma20** [comment]
Have you sourced the right Vivado version? E.g. "source /tools/Xilinx/Vivado/2020.1/settings64.sh"

For Vivado 2020.2 you can use the "feature/vitis_hls" branch of FINN

**Reply by MehdiRh17** [reply]
Hello and thanks for your reply. I assume that I should run these commands respectively, right?
source /tools/Xilinx/Vivado/2020.1/settings64.sh, which add vivado to my path, 
then : export FINN_XILINX_PATH=/opt/Xilinx/Vivado/2019.1/      and       
export FINN_XILINX_VERSION=2019.   according to the main documentation. After running  mentioned commmands, I still get the warning that setting.64.sh cannot be find and finally I cannot run the container. I really appreciate if if you can help me with this problem. Thanks alot

**Reply by MehdiRh17** [reply]
Here is also the full error message (the container is running though) : 

 WARNING: Unable to find /settings64.sh
WARNING: Functionality dependent on Vitis will not be available.
WARNING: If you need Vitis, ensure VITIS_PATH is set correctly and mounted into the Docker container.
WARNING: Unable to find /settings64.sh
WARNING: Functionality dependent on Vivado will not be available.
WARNING: If you need Vivado, ensure VIVADO_PATH is set correctly and mounted into the Docker container.

**Reply by nhma20** [reply]
It looks like your paths are still not set up correctly. Make sure your paths point to the right place (does the file "/tools/Xilinx/Vivado/2019.1/settings64.sh" exist?) and correspond to the tools version installed on your system. So for your Vivado 2019.1 installation that might look like:

source /tools/Xilinx/Vivado/2019.1/settings64.sh
export FINN_XILINX_PATH=/tools/Xilinx
export FINN_XILINX_VERSION=2019.1
export VIVADO_PATH=/tools/Xilinx/Vivado/2019.1

You need to execute the above in the terminal before you run the docker. For convenience, I have put these in my ~/.bashrc file such that they are executed automatically every time I open a new terminal.

**Comment by MehdiRh17** [comment, marked as answer]
Thank you very much. Solved the issue.
